src/main/java/org/trebol/jpa/services/predicates/command/PredicateCommand.java

```diff
@@ -0,0 +1,71 @@
+package org.trebol.jpa.services.predicates.command;
+
+import com.querydsl.core.BooleanBuilder;
+import com.querydsl.core.types.Predicate;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.util.CollectionUtils;
+import org.trebol.jpa.entities.QProductCategory;
+import org.trebol.jpa.services.IProductCategoryTreeResolver;
+
+import java.util.List;
+import java.util.Map;
+
+public class PredicateCommand {
+  private static final String ID = "id";
+  private static final String CODE = "code";
+  private static final String NAME = "name";
+  private static final String NAME_LIKE = "nameLike";
+  private static final String PARENT_CODE = "parentCode";
+  private static final String PARENT_ID = "parentId";
+  private static final String ROOT_ID = "rootId";
+  private static final String ROOT_CODE = "rootCode";
```

Comment:
Is this `PredicateCommand` can only be used for the `ProductCategoriesPredicateJpaServiceImpl` ?

Reply:
Yes. I just refactor the switch case since its tag as cyclomatic issue in sonar and convert to command pattern